src/Workspaces/Core/Portable/Workspace/Solution/RecoverableTextAndVersion.RecoverableText.cs

```diff
@@ -23,8 +24,11 @@ internal sealed partial class RecoverableTextAndVersion
     private sealed partial class RecoverableText
     {
         // enforce saving in a queue so save's don't overload the thread pool.
-        private static Task s_latestTask = Task.CompletedTask;
-        private static readonly NonReentrantLock s_taskGuard = new();
+        private static readonly AsyncBatchingWorkQueue<(RecoverableText recoverableText, SourceText sourceText)> s_saveQueue =
+            new(TimeSpan.Zero,
+                SaveAllAsync,
+                AsynchronousOperationListenerProvider.NullListener,
```

Comment:
All asynchronous operations must be tracked. I recommend `FeatureAttribute.Workspace` for this queue.

Reply:
The async work was not tracked before. Async work needs to be tracked when there is a need for tests to be able to say "I need this with to be done before I can properly test things"

Reply:
> The async work was not tracked before.

Yes, this is correct.

> Async work needs to be tracked when there is a need for tests to be able to say "I need this with to be done before I can properly test things"

In the abstract, this is true. In practice, this is not a decision for the code to make at the call site. It's a hard rule because if a case is ever missed, finding it is basically impossible.

Reply:
I agree with both of you. This should probably specify a listener, but shouldn't be blocked on that as it's not changed behavior.